src/components/chart/state-history-charts.ts

```diff
@@ -152,6 +162,11 @@ class StateHistoryCharts extends LitElement {
     return !(changedProps.size === 1 && changedProps.has("hass"));
   }
 
+  yWidthChanged(e) {
+    this._childYWidths[e.detail.chartIndex] = e.detail.value;
```

Comment:
Shouldn't `_childYWidths` be reset whenever `historyData` includes different entities? Or at least has a different length, as if the number of items got smaller, the higher indexes will not be updated?

Reply:
I haven't noticed this problem in testing because it seems like the whole state-history-charts instance is destroyed and recreated when the entities change. Anytime I add or remove an entity, in the first call to yWidthChanged, this._childYWidths is always empty, and then each chart updates it for the first time from there.

Reply:
In most cases that will happen probably (because we render a spinner in between), but it is possible that it could be used without recreating the element

Reply:
I may be speaking in hypotheticals since I don't know if this situation could happen, but in theory if we did reset the array on a change, I don't know that existing entities would necessarily perform their update again, if their width hadn't changed. 

E.g. if we remove the 4th chart, charts 1-3 might not re-report their new width, so I don't think I just want to reset. 

Maybe answer could be in an updated() function, truncate childYWidth if it is longer than length of combinedItems? Though seems I could not really exercise this currently, so not sure if it is moot.

Reply:
I added the `updated` function to state-history-charts to try to address this, but admittedly I was never able to trigger a breakpoint inside the innermost if statement by adding or removing entities, neither in the history card, or in the history panel from the sidebar, so I couldn't actually confirm that it does what is intended.